drivers/opt3001/opt3001.c

```diff
@@ -0,0 +1,194 @@
+    reg &= ~OPT3001_REGS_CONFIG_ME;
+    reg &= ~OPT3001_REGS_CONFIG_FC_MASK;
+
+    reg = htons(reg);
+
+    if (i2c_write_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_init: Error setting device configuration\n");
+        return -OPT3001_ERROR_BUS;
+    }
+    i2c_release(DEV_I2C);
+
+    return OPT3001_OK;
+}
+
+int opt3001_reset(const opt3001_t *dev)
+{
+    uint16_t reg = OPT3001_CONFIG_RESET;
+    reg |= OPT3001_REGS_CONFIG_CT(OPT3001_CONVERSION_TIME);
+    reg = htons(reg);
+
+    /* Acquire exclusive access to the bus. */
+    i2c_acquire(DEV_I2C);
+    if (i2c_write_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        return -OPT3001_ERROR_BUS;
+    }
+    i2c_release(DEV_I2C);
+    return OPT3001_OK;
+}
+
+int opt3001_set_active(const opt3001_t *dev)
+{
+    uint16_t reg;
+
+    i2c_acquire(DEV_I2C);
+
+    if (i2c_read_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_init: Error reading BUS!\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    reg = htons(reg);
+    reg |= OPT3001_REGS_CONFIG_MOC(OPT3001_CONFIG_M_CONTINUOUS);
+
+    reg = htons(reg);
+    if (i2c_write_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_init: Error setting device configuration\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    i2c_release(DEV_I2C);
+    return OPT3001_OK;
+}
+
+int opt3001_read(const opt3001_t *dev, uint16_t *crf, uint16_t *rawl)
+{
+    uint16_t reg;
+
+    i2c_acquire(DEV_I2C);
+
+    /* wait for the conversion to finish */
+    if (OPT3001_CONVERSION_TIME) {
+        xtimer_usleep(OPT3001_CONVERSION_TIME_LONG);
+    } else{
+        xtimer_usleep(OPT3001_CONVERSION_TIME_SHORT);
+    }
+
+    if (i2c_read_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_init: Error reading BUS!\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    *crf = htons(reg) & OPT3001_REGS_CONFIG_CRF;
+
+    if (!(*crf)) {
+        i2c_release(DEV_I2C);
+        LOG_DEBUG("opt3001_read: conversion in progress!\n");
+        return -OPT3001_ERROR;
+    }
+
+    if (i2c_read_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_RESULT, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_init: Error reading BUS!\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    *rawl = htons(reg);
+
+    i2c_release(DEV_I2C);
+    return OPT3001_OK;
+}
+
+void opt3001_convert(int16_t rawl, float *convl)
+{
+  uint16_t mantissa;
+  uint8_t exponent;
+
+  exponent = OPT3001_REGS_REG_EXPONENT(rawl);
+  mantissa = OPT3001_REGS_REG_MANTISSA(rawl);
+
+  *convl = 0.01 * pow(2, exponent) * mantissa;
+}
+
+int opt3001_read_lux(const opt3001_t *dev, int16_t *convl)
+{
+    uint16_t crf;
+#if (!OPT3001_USE_RAW_VALUES)
```

Comment:
What is the idea behind the function returning different things depending on the config?